lib/date_range.rb

```diff
@@ -0,0 +1,39 @@
+require 'date'
+
+module Hotel
+  class InvalidDateRangeError < StandardError
+  end
+
+  class DateRange
+    attr_reader :start_date, :end_date, :number_of_nights
+
+    def initialize(start_date, end_date)
+      @start_date = start_date
+      @end_date = end_date
+      # check_valid
+      @number_of_nights = (end_date - start_date).to_i
+      check_valid
+
+    end
+
+    def check_valid
+      raise InvalidDateRangeError.new("Invalid Date Range.") if @number_of_nights <= 0
+    end
+
+    def nights_arr
+      nights_arr = []
+      @number_of_nights.times do |i|
+        nights_arr << (@start_date + i)
+      end
+      return nights_arr
+    end
+
+    def include?(check_date)
+      nights_arr.each do |date|
+        return true if date == check_date
+      end
```

Comment:
I like this method! Checking whether a date range includes a date definitely seems like it's `DateRange`'s responsibility